src/datajudge/constraints/row.py

```diff
@@ -36,6 +38,8 @@ def test(self, engine: sa.engine.Engine) -> TestResult:
 
 class RowEquality(Row):
     def get_factual_value(self, engine: sa.engine.Engine) -> Tuple[int, int]:
+        if self.ref is None or self.ref2 is None:
```

Comment:
This is a bit annoying. The `Row` constraints are always between several data sources and thereby `ref` and `ref2` should never be None. One can easily see in their initialization function, that the inputs to the initialization have according type hints. Yet, mypy doesn't seem to pick this up.

Reply:
Don't know a better way.

On another note, the line before has a Tuple return type instead of tuple.

Reply:
Great catch!